ipyleaflet/leaflet.py

```diff
@@ -1170,6 +1170,8 @@ def add(self, layer):
 
         if isinstance(layer, dict):
             layer = basemap_to_tiles(layer)
+        elif hasattr(layer, 'as_leaflet_layer'):
```

Comment:
Could you add a comment saying that it's specific to `localtileserver`?

Reply:
We might want to generalize this interface? I think it could make sense outside `localtileserver` too, for instance in `xarray-leaflet`.

Reply:
Yeah that makes sense!

Reply:
My intention is definitely for this to be generally applicable. 

I'd love to see libraries like `shapely` have an `as_leaflet_layer` too (or perhaps I'll just monkey path that method) to make my life easier